lib/generators/templates/active_job/active_job.rb

```diff
@@ -0,0 +1,14 @@
+require "active_job/logging"
+require "active_job/log_subscriber"
+
+ActiveSupport::Notifications.unsubscribe("enqueue.active_job")
+
+module ActiveJob
+  module Logging
+    class EnqueueLogSubscriber < LogSubscriber
+      define_method :enqueue, instance_method(:enqueue)
```

Comment:
I think this doesn't work in Rails 7.1.

Have you tested the generator "manually" on a new application?

Reply:
Thank you for the comment @crackofdusk ! It has made me investigate more and try the generator in a new app. 
It looks like it’s working, but please let me know if you think otherwise. 

I took the following steps to test it:

- I created a new Rails 7.1.2 App
- I added the generator to this app.
- I added the template `active_job.rb`
- I ran in my terminal `bin/rails generate jobs`

- This has generated the following
    
    ```ruby
    gemfile  sidekiq
             run  bundle install from "."
    Bundle complete! 16 Gemfile dependencies, 85 gems now installed.
    Use `bundle info [gemname]` to see where a bundled gem is installed.
          create  config/initializers/active_job.rb
    Add default Procfile.dev
          create  Procfile.dev
    ```
- I can see the `sidekiq` gem in the Gemfile, the `active_job` file in initializers, and the `Procfile` with the expected configuration.
- Then I create a job - `MyJob`

I modified the initializer to: 
```
require "active_job/logging"
require "active_job/log_subscriber"


# ActiveSupport::Notifications.unsubscribe("enqueue.active_job")
Rails.logger.info "Custom Active Job initializer is running"

module ActiveJob
  module Logging
    class EnqueueLogSubscriber < LogSubscriber
      puts "ActiveJob::Logging::EnqueueLogSubscriber"
      puts "I'm running"
      define_method :enqueue, instance_method(:enqueue)
    end
  end
end

ActiveJob::Logging::EnqueueLogSubscriber.attach_to(:active_job)
```

I ran `rails console` and I can see:
```
ActiveJob::Logging::EnqueueLogSubscriber
I'm running
```
and when I run the job `MyJob.perform_later`, I can see in the logs:
```
Custom Active Job initializer is running
[ActiveJob] Enqueued MyJob (Job ID: 02a76977-c7a1-47c6-8366-787c097b83f4) to Sidekiq(default)
[ActiveJob] ↳ (irb):1:in `<main>'
[ActiveJob] Enqueued MyJob (Job ID: 02a76977-c7a1-47c6-8366-787c097b83f4)
```

Let me know what you think 😃

Reply:
Thank you for testing! I had a problem with this patch when upgrading an application from Rails 7.0 to 7.1. The error could have been caused by something else. For me it's enough if this generator works a new Rails application.